using_restricted_array.rb

```diff
@@ -8,28 +8,62 @@
 
 ## Calculates the length of the restricted integer array_size
 def length(array)
-  puts "NOT IMPLEMENTED"
+  array_length = 0
+
+  until array[array_length] == nil do
+    array_length += 1
+  end
+
+  return array_length
+
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  i = 0
+
+  until i == length(array) do
+    puts array[i]
+    i += 1
+  end
+
 end
 
 # Reverses the values in the integer array
 def reverse(array, length) # Ruby
-  puts "NOT IMPLEMENTED"
+  pointer_1 = 0
+  pointer_2 = length - 1
+  temp = nil
+
+  until pointer_1 > pointer_2 do
+    temp = array[pointer_1]
+    array[pointer_1] = array[pointer_2]
+    array[pointer_2] = temp
+
+    pointer_1 += 1
+    pointer_2 -= 1
+  end
+
+  return array
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+  i = 0
+
+  until i == length do
+    if array[i] == value_to_find
+      return true
+    end
+    i += 1
+  end
+  return false
 end
 
 # Sorts the array in ascending order.
 def sort(array, length)
-  puts "NOT IMPLEMENTED"
+puts "Sorry, Shruti! I didn't have time but will finish this this week."

```

Comment:
Np. Send me a slack message if you'd like me to review it when you complete it. Or you can cross check with https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/classroom/Sorting.md